attest/tpm_linux.go

```diff
@@ -408,6 +422,7 @@ func (t *TPM) MintAIK(opts *MintOptions) (*Key, error) {
 			KeyEncoding:       KeyEncodingParameterized,
```

Comment:
Should this be a different `KeyEncoding`?

Reply:
(such as `KeyEncodingEncrypted`)

Reply:
Probably yes, should we also drop the check for KeyEncoding when loading?

Reply:
The intent of the `KeyEncoding` field is to describe how the serialized key is represented. `KeyEncodingParameterized` should indicate to `LoadKey` that the key is a primary key that needs to be generated to get the key handle. `KeyEncodingEncrypted` symbolizes that the encrypted private blob is present, and the key needs to be loaded to get the handle.

The aim of this was to support keys that were generated with an older version of the library.

So either:

1. We should change the check of KeyEncoding to `KeyEncodingEncrypted`, dropping support for loading the old key format
2. We should use a switch to support loading both.

My preference is the second, but its up to you.

Reply:
I've already made the change for the 1st option